src/licensedcode/detection.py

```diff
@@ -118,12 +153,90 @@ class LicenseDetection:
         )
     )
 
+    matches = attr.ib(
+        #repr=matches_compact_repr,
+        default=attr.Factory(list),
+        metadata=dict(
+            help='List of license matches combined in this detection.'
+        )
+    )
+
+    @classmethod
+    def from_matches(cls, matches):
+        """
+        Return a LicenseDetection created out of `matches` list of LicenseMatch
```

Comment:
Less is more here: let's not expose too much of the sausage making in the docstring.
```suggestion
        Return a LicenseDetection created out of `matches` list of LicenseMatch.
```

Reply:
Added.